velox/functions/prestosql/ArrayIntersectExcept.cpp

```diff
@@ -269,29 +271,143 @@ class ArrayIntersectExceptFunction : public exec::VectorFunction {
+        if (decodedLeftElements->isNullAt(i)) {
+          // Arrays overlap skips null values.
+          continue;
+        }
+        if (rightSet.set.count(decodedLeftElements->valueAt<T>(i)) > 0) {
+          // Found and overlapping element. Add to result set.
+          resBoolVector->set(rowId, true);
+          return;
+        }
+      }
+      // If none of them is found in the rightSet, set false for current row
+      // indicating there are no overlapping elements with rightSet.
+      resBoolVector->set(rowId, false);
+    };
+
+    if (constantSet_.has_value()) {
+      rows.applyToSelected(
+          [&](vector_size_t row) { processRow(row, *constantSet_); });
+    }
+    // General case when no arrays are constant and both sets need to be
+    // computed for each row.
+    else {
+      exec::LocalDecodedVector rightDecoder(context, *right, rows);
+      exec::LocalDecodedVector rightElementsDecoder(context);
+      auto decodedRightElements = getDecodedElementsFromArrayVector(
+          rightDecoder, rightElementsDecoder, rows);
+      SetWithNull<T> rightSet;
+
+      rows.applyToSelected([&](vector_size_t row) {
+        auto idx = rightDecoder.get()->index(row);
+        auto baseRightArray = rightDecoder.get()->base()->as<ArrayVector>();
+        generateSet<T>(baseRightArray, decodedRightElements, idx, rightSet);
+        processRow(row, rightSet);
+      });
+    }
+  }
+
+ private:
+  // If one of the arrays is constant, this member will store a pointer to the
+  // set generated from its elements, which is calculated only once, before
+  // instantiating this object.
+  std::optional<SetWithNull<T>> constantSet_;
 
-void validateType(const std::vector<exec::VectorFunctionArg>& inputArgs) {
+  // If there's a `constantSet`, whether it refers to left or right-hand side.
+  const bool isLeftConstant_{false};
+}; // class ArraysOverlapFunction
+
+void validateType(
+    const std::vector<exec::VectorFunctionArg>& inputArgs,
+    const std::string name,
+    const vector_size_t expectedArgCount) {
   VELOX_USER_CHECK_EQ(
-      inputArgs.size(), 2, "array_intersect requires exactly two parameters");
+      inputArgs.size(),
+      expectedArgCount,
+      "{} requires exactly {} parameters",
+      name,
+      expectedArgCount);
 
   auto arrayType = inputArgs.front().type;
   VELOX_USER_CHECK_EQ(
       arrayType->kind(),
       TypeKind::ARRAY,
-      "array_intersect requires arguments of type ARRAY");
+      "{} requires arguments of type ARRAY",
+      name);
 
   for (auto& arg : inputArgs) {
     VELOX_USER_CHECK(
         arrayType->kindEquals(arg.type),
-        "array_intersect function requires all arguments of the same type: {} vs. {}",
+        "{} function requires all arguments of the same type: {} vs. {}",
+        name,
         arg.type->toString(),
         arrayType->toString());
   }
 }
 
+template <typename T>
+void validateConstVecAndGenerateSet(
```

Comment:
do not abbreviate: validateConstantVectorAndGenerateSet

Based on usage of this function, it would be more convenient if it returned SetWithNull<T> instead of taking it as an output parameter.